qiita_pet/static/js/networkVue.js

```diff
@@ -713,22 +713,12 @@ Vue.component('processing-graph', {
      **/
     addJobNodeToGraph: function(job_info) {
       let vm = this;
-      // Fun fact - although it seems counterintuitive, in vis.Network we
-      // first need to add the edge and then we can add the node. It doesn't
-      // make sense, but it is how it works
-      $(job_info.inputs).each(function(){
-        vm.edges_ds.add({id: vm.edges_ds.length + 1, from: this, to: job_info.id});
-      });
-      var node_info = vm.colorScheme.in_construction;
-      vm.nodes_ds.add({id: job_info.id, shape: node_info['shape'], group: "job", shape: 'dot', label: formatNodeLabel(job_info.label), color: node_info, status: 'in_construction'});
-      $(job_info.outputs).each(function(){
-        var out_name = this[0];
-        var out_type = this[1];
-        var n_id = job_info.id + ":" + out_name;
-        vm.edges_ds.add({id: vm.edges_ds.length + 1, from: job_info.id, to: n_id });
-        vm.nodes_ds.add({id: n_id, shape: 'dot', label: formatNodeLabel(out_name + "\n(" + out_type + ")"), group: "type", name: out_name, type: out_type});
-      });
-      vm.network.redraw();
+
+      vm.new_job_info = {
```

Comment:
Am I reading this right, do we no longer need to manually add edges and nodes?

Reply:
That is correct, now we reload the network (vm.updateGraph();) this assures that we get the final position of the new nodes and simplifies the code.

Reply:
Got it, thanks!